plugin/azure/azure.go

```diff
@@ -346,7 +346,14 @@ func (h *Azure) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg)
 		return dns.RcodeServerFailure, nil
 	}
 
-	w.WriteMsg(m)
+	err := w.WriteMsg(m)
```

Comment:
we don't really care if this fails, there's nothing what we can do about, logging it will allows for easy DoS-ing of whatever these logs end up in. Please remove.

original change lgtm

Reply:
Thanks, and good point - will do. Just to check - should we log as Debug? I only ask because the logs that were written gave the impression that a response was returned to the client, when it wasn't.

I'm happy either way and will adjust the PR once you can confirm.

Reply:
hmm, true point. We can save the err and return it from the plugin, then it will be logged by errors plugin, but that would be a system wide change. 
Apart from we can't pack a message because of an issue like this, failing a write to a client is not important. We _could_ check for dns: type errors and only log those...

Reply:
I think what you propose makes sense - just return any error directly without logging. I just tried that and, as you say, it causes the errors plugin to pick it up and I see a reasonable log entry:
```
[ERROR] plugin/errors: 2 shop-job-adapter.sxdb-blake-dev-neu.azdevint.dh. AAAA: dns: domain must be fully qualified 
```